### ISSUE TITLE
Check-PSGalleryApiAvailability fails when network is available

### ISSUE THREAD

**Issue by iricigor** [PowerShell/PowerShellGetv2#270]
Function that checks for PS Gallery API availability ([link to dev branch here](https://github.com/PowerShell/PowerShellGet/blob/development/PowerShellGet/private/functions/Check-PSGalleryApiAvailability.ps1)) is performing these three steps:

```PowerShell
Test-Connection -ComputerName $microsoftDomain
Test-NetConnection -ComputerName $microsoftDomain
[System.Net.NetworkInformation.NetworkInterface]::GetIsNetworkAvailable()
```

## Expected Behavior

If any of those commands returns `$true` check should be successful. 

## Current Behavior

If some command throws an error,  entire check will fail. Commands after failing one are not executed.

## Possible Solution

We can catch exceptions thrown from these commands and proceed checking with next commands until we find connection or we do all three checks.

Something like:

```PowerShell
if (command1available) {
  try {$connected = Do-Check1}
  catch {$connected = $false}
}

if ( -not $connected -and (command2available)) {
  try {$connected = Do-Check2}
  catch {$connected = $false}
}

if ( -not $connected -and (command3available)) {
  try {$connected = Do-Check3}
  catch {$connected = $false}
}
```

Or, another option would be to simply add `-ErrorAction SilentlyContinue` to each check as `$connected` is set to `$false` in advance.

## Steps to Reproduce (for bugs)

I encountered this issue while playing with Azure Functions.
Exact steps to identify this issue is to simply run these commands inside custom PowerShell Azure Function

```PowerShell
Find-Module PowerShellGet
Install-Module PowerShellGet -Scope CurrentUser
```

First one will work fine, second one will fail on described error.

To further narrow down the error, you may run these commands:
```PowerShell
[System.Net.NetworkInformation.NetworkInterface]::GetIsNetworkAvailable()
Test-NetConnection -ComputerName 'www.microsoft.com' -InformationLevel Quiet
```

First one will return `$true`, while second one will throw an exception 

```
[Error] Test-Connection : No such interface supported
```

Also, running `Get-NetAdapter` command inside of the Azure Function will give you:

```
Get-NetAdapter : Cannot connect to CIM server. Access is denied
```

## Your Environment

```PowerShell
$PSVersionTable
```

```
Name                           Value                                                                                    

----                           -----                                                                                    

PSVersion                      5.1.14393.2125                                                                           

PSEdition                      Desktop                                                                                  

PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                  

BuildVersion                   10.0.14393.2125                                                                          

CLRVersion                     4.0.30319.42000                                                                          

WSManStackVersion              3.0                                                                                      

PSRemotingProtocolVersion      2.3                                                                                      

SerializationVersion           1.1.0.1

```

```PowerShell
Get-Module -ListAvailable PowerShellGet,PackageManagement
```

```
Directory: D:\Program Files\WindowsPowerShell\Modules





ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-Packag...

Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}  





    Directory: D:\Program Files (x86)\WindowsPowerShell\Modules





ModuleType Version    Name                                ExportedCommands                                              

---------- -------    ----                                ----------------                                              

Binary     1.0.0.1    PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-Packag...

Script     1.0.0.1    PowerShellGet                       {Install-Module, Find-Module, Save-Module, Update-Module...}
```

```PowerShell
Get-PackageProvider -ListAvailable
```

```
Name                     Version          DynamicOptions                                                                

----                     -------          --------------                                                                

msi                      3.0.0.0          AdditionalArguments                                                           

msu                      3.0.0.0                                                                                        

PowerShellGet            1.0.0.1          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, I...

Programs                 3.0.0.0          IncludeWindowsInstaller, IncludeSystemComponent
```

## P.S.

I can do PR if needed, but I'd like to hear opinions first. Maybe this check is not needed at all?

**Comment by bmanikm** [comment]
@iricigor thanks for the proposed solution! If possible, please submit a PR too.

**Comment by iricigor** [comment]
Seems like `Test-Connection` throws an error even with `-ErrorAction SilentlyContinue`. So, I'd rather go with solution with try catch on all three commands.

But that, raises another question. `catch {$connected = $false}` is not really needed, as `$connected` is set to `$false` in advance. So, it is enough to have empty catch block, like `catch {}` all three times. I will just add comments after them ([like this](https://github.com/iricigor/PowerShellGet/commit/43de12df4393d078983d0ab0c73b1fb371e6e621)). 

That's OK, or someone has better suggestion?

**Comment by bmanikm** [comment]
@iricigor I am fine with the empty catch blocks. Did you get a chance to try with `-ErrorAction Ignore`?

**Comment by iricigor** [comment]
Yes, I tried -EA Ignore. It had the same effect at the end.

**Comment by BladeFireLight** [comment]
I see that the proposed solution is in place. 

However I question the whole process.

What does the ability to ping www.microsoft.com have to do with checking a URL's API. 

In our environment these checks add 45 seconds to every command that uses this function as they timeout trying to check for a nonexistent internet connection. 

If your testing the API itself, the only thing that matters is, can you talk the required protocol to the target URL. 

Why not just test that?

**Comment by SydneyhSmith** [comment]
Closing as I believe this was resolved with PR #374 will re-open if this is still an issue
